django/contrib/admin/options.py

```diff
@@ -1505,15 +1506,32 @@ def get_changeform_initial_data(self, request):
         Get the initial form data from the request's GET params.
         """
         initial = dict(request.GET.items())
+        formset_data = defaultdict(lambda: defaultdict(defaultdict))
+        formset_query_regex = r'inlinemode_set-(\d+)-(\d+)-(.*)'
         for k in initial:
             try:
+                match_obj = re.match(formset_query_regex, k)
+                if match_obj:
+                    inline_model_index = int(match_obj.group(1))
+                    inline_model_form_index = int(match_obj.group(2))
+                    inline_model_form_fieldname = match_obj.group(3)
+                    model_field = self.inlines[inline_model_index].model._meta.get_field(inline_model_form_fieldname)
+                    if isinstance(model_field, models.ManyToManyField):
+                        formset_data[inline_model_index][inline_model_form_index][inline_model_form_fieldname] \
+                            = initial.get(k).split(",")
+                    else:
+                        formset_data[inline_model_index][inline_model_form_index][inline_model_form_fieldname] \
+                            = initial.get(k)
+
                 f = self.model._meta.get_field(k)
             except FieldDoesNotExist:
                 continue
+            except IndexError:
+                continue
             # We have to special-case M2Ms as a list of comma-separated PKs.
             if isinstance(f, models.ManyToManyField):
                 initial[k] = initial[k].split(",")
-        return initial
+        return initial, formset_data
```

Comment:
This is changing the return type of `get_changeform_initial_data()` which is a documented function.

Documentation changes are missing. This may well also need a deprecation period unless there is an alternative way of solving this.

Reply:
@carltongibson  @pope1ni  What I can do is to create a separate helper function as mentioned by @carltongibson  in the previous review and then call the function directly for the _changeform_view function. This will prevent the need to deprecate. What do you think ?